Sources/Calendar.php

```diff
@@ -219,18 +219,31 @@ function CalendarMain()
 function CalendarPost()
 {
 	global $context, $txt, $user_info, $sourcedir, $scripturl;
-	global $modSettings, $topic, $smcFunc;
+	global $modSettings, $topic, $smcFunc, $settings;
 
 	// Well - can they?
 	isAllowedTo('calendar_post');
 
-	// We need this for all kinds of useful functions.
+	// We need these for all kinds of useful functions.
 	require_once($sourcedir . '/Subs-Calendar.php');
+	require_once($sourcedir . '/Subs.php');
 
 	// Cast this for safety...
 	if (isset($_REQUEST['eventid']))
 		$_REQUEST['eventid'] = (int) $_REQUEST['eventid'];
 
+	// We want a fairly compact version of the time, but as close as possible to the user's settings.
+	if (preg_match('~%[HkIlMpPrRSTX](?:[^%]*%[HkIlMpPrRSTX])*~', $user_info['time_format'], $matches) == 0 || empty($matches[0]))
```

Comment:
I believe that SMF always wants to wrap if / else with brackets, no? Been a while